lib/inferno/dsl/auth_info.rb

```diff
@@ -168,6 +168,143 @@ def add_to_client(client)
 
         client.set_bearer_token(access_token)
       end
+
+      # @private
+      def need_to_refresh?
+        return false if access_token.blank? || (auth_type != 'backend_services' && refresh_token.blank?)
+
+        return true if expires_in.blank?
+
+        issue_time.to_i + expires_in.to_i - DateTime.now.to_i < 60
+      end
+
+      # @private
+      def able_to_refresh?
+        (auth_type == 'backend_services' && token_url.present?) || (refresh_token.present? && token_url.present?)
```

Comment:
Let's make `backend_services?` etc. helper methods for checking the auth type. `token_url.present? && (backend_services? || refresh_token.present?)`.